eclipse-collections/src/main/java/org/eclipse/collections/impl/set/mutable/MultiReaderSetAdapter.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (c) 2020 Goldman Sachs and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * and Eclipse Distribution License v. 1.0 which accompany this distribution.
+ * The Eclipse Public License is available at http://www.eclipse.org/legal/epl-v10.html
+ * and the Eclipse Distribution License is available at
+ * http://www.eclipse.org/org/documents/edl-v10.php.
+ */
+
+package org.eclipse.collections.impl.set.mutable;
+
+import java.util.Set;
+import java.util.concurrent.locks.ReadWriteLock;
+import java.util.concurrent.locks.ReentrantReadWriteLock;
+
+import org.eclipse.collections.api.set.MultiReaderSet;
+import org.eclipse.collections.api.set.MutableSet;
+
+public final class MultiReaderSetAdapter<T> extends AbstractMultiReaderSet<T>
+{
+    MultiReaderSetAdapter(MutableSet<T> newDelegate, ReadWriteLock newLock)
```

Comment:
package private constructor in public class. Not sure if you want this to be public or package private but the two visibilities should agree

Reply:
I did that to be similar to how `SetAdapter` was defined: the thing is the class must be public so that `MultiReaderSetAdapter.adapt()` can be called from anywhere. Should I make the constructor private maybe? Maybe in the case of SetAdapter, it is package private so that it is accessible for tests, but apart from that I don't know.

Reply:
It's to force the use of the adapt() method which has instanceof checks or something like that. Seems fine.